extendr-api/src/wrapper/externalptr.rs

```diff
@@ -109,14 +109,14 @@ impl<T> Deref for ExternalPtr<T> {
     }
 }
 
-impl<T> DerefMut for ExternalPtr<T> {
+impl<T: 'static> DerefMut for ExternalPtr<T> {
     /// This allows us to treat the Robj as if it is the mutable type T.
     fn deref_mut(&mut self) -> &mut Self::Target {
         self.addr_mut()
     }
 }
 
-impl<T> ExternalPtr<T> {
+impl<T: 'static> ExternalPtr<T> {
```

Comment:
I'm worried about these bounds here—though I'm not entirely sure what it implies. Does the static bound mean that the _type_ or the struct must be alive for the duration of the process?

Reply:
The issue is, that the bounds come from `std::any::TypeId::of`. So they are not optional. But they were present before, I just removed them, when we didn't use the `std::any::type_name` or the other `TypeId` thing.

I'm also not sure what `T: 'a` means. I don't think it means anything in particular here either.

Reply:
I asked about it in the Rust discord and I got many interesting answers.
link: https://discord.com/channels/442252698964721669/443150878111694848/1293353783178428417

> It basically means that the type doesn't hold any borrows that are non-'static lifetime
This includes any type that doesn't borrow at all

> Or in practical terms, it means that you can hold on to that value as long as you like without fear that anything it borrows might go away
Which for example is important for threads, since they can run long after the calling function has returned

> for TypeId specifically, it's important because lifetimes are erased before types are given ids 
so to prevent you from freely casting &'a T to &'static T, Any and TypeId require 'static

Reply:
I believe what it simply means, is that TypeId contains the information of lifetimes as well as types, and thus, it becomes part of the typeid.

Reply:
I followed up with this conversation and it makes sense. Thank you!